### PR TITLE
Log event IDs before sending to Sentry

### PR DESCRIPTION
**Pull request by chrisglass**
This should allow people investigating failures to find things faster/more
easily.

Also refactored the code a little, reduced duplication and ran "lein fmtclj".

### CODE REVIEW

**Comment by chrisglass** [comment]
(this branch needs https://github.com/exoscale/raven/pull/20 to be landed and released first)

File: src/spootnik/reporter.clj [hunk 1]
```diff
@@ -246,19 +246,14 @@
...
-      (try
-        (raven/capture! {:http raven} (:dsn sentry) e tags)
-        (catch Exception e
-          (error e "could not send capture")))))
+    (when-let [event-id (raven/sentry-uuid!)] raven
+              (error e (str "captured exception as sentry event: " event-id))
+              (try
+                (raven/capture! {:http_client raven :event_id event-id} (:dsn sentry) e tags)
+                (catch Exception e
+                  (error e "could not send capture for event: " event-id)))))
```
Comment (greut) [inline src/spootnik/reporter.clj]: I'd prefer something along those lines

```clj
(try
  (when-let [event-id (raven/capture! ... e tags)]
    (error e (str "captured exception" event-id))
  (catch ...))
```

Comment (greut) [reply src/spootnik/reporter.clj]: even if, I don't see the reason for the `(try (catch ...))` in the first place

Comment (chrisglass) [reply src/spootnik/reporter.clj]: Agreed.

Done :+1:

Comment (chrisglass) [reply src/spootnik/reporter.clj]: (I removed the try/catch as well)

File: src/spootnik/reporter.clj [hunk 2]
```diff
@@ -246,19 +246,10 @@
...
+    (capture! this e {}))
   (capture! [this e tags]
-    (error e "captured exception")
-    (when raven
-      (try
-        (raven/capture! {:http raven} (:dsn sentry) e tags)
-        (catch Exception e
-          (error e "could not send capture")))))
+    (when-let [event-id (raven/capture! {:http_client raven} (:dsn sentry) e tags)] raven
+              (error e (str "captured exception as sentry event: " event-id))))
```
Comment (greut) [inline src/spootnik/reporter.clj]: you'll lost a raven :bird:

Comment (chrisglass) [reply src/spootnik/reporter.clj]: Oops! It looks like when-let does not do what I thought it did.

Thanks! :+1: 

Done.

**Pull request merged**